tests/ui/dyn-star/param-env-region-infer.rs

```diff
@@ -1,9 +1,10 @@
-// revisions: current next
-// Need `-Zdeduplicate-diagnostics=yes` because the number of cycle errors
-// emitted is for some horrible reason platform-specific.
-//[next] compile-flags: -Ztrait-solver=next -Zdeduplicate-diagnostics=yes
+// revisions: current
 // incremental
 
+// FIXME(-Ztrait-solver=next): THis currently results in unstable query results:
+// `normalizes-to(opaque, opaque)` changes from `Maybe(Ambiguous)` to `Maybe(Overflow)`
+// once the hidden type of the opaque is already defined to be itself.
+
```

Comment:
I looked at this for a bit and I think that we should look into preventing opaques in their defining scope to opaques in their defining scope. Not exactly sure how this should work though. Feels like a complex issue so I've avoided it for now.

Reply:
Oh, yeah, yikes lol.